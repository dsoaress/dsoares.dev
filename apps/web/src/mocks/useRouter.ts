export default jest.mock('next/router', () => ({
  useRouter: () => ({
    locale: 'en',
    locales: ['en', 'es', 'pt'],
    asPath: '/'
  })
}))
