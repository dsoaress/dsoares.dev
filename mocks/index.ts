export * from './nextHead'
export * from './nextImage'
export * from './projects'
export * from './useData'
export * from './useRouter'
