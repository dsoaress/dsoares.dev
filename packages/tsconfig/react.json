{
  "$schema": "https://json.schemastore.org/tsconfig",
  "display": "React",
  "extends": "./base.json",
  "compilerOptions": {
    "jsx": "react-jsx",
    "lib": ["ES2015"],
    "module": "ESNext",
    "target": "es6"
  }
}
