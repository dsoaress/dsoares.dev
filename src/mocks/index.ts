export * from './nextHead'
export * from './nextImage'
export * from './projects'
export * from './useCurrentTrack'
export * from './useData'
export * from './useRouter'
