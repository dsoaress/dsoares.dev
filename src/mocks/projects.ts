import type { Project } from '@/types'

const tags = 'tag a, tag b, tag c'

export const projects: Project[] = [
  {
    title: 'Project 1',
    description: {
      en: 'Project 1 description',
      es: 'Descripción del proyecto 1',
      pt: 'Descrição do projeto 1'
    },
    tags,
    repo: 'project-1',
    repositoryUrl: 'https://github.com/johndoe/project-1',
    cover: '/assets/projects/project-1.png',
    coverBlurDataURL: '',
    forks: 5,
    stars: 10
  },
  {
    title: 'Project 2',
    description: {
      en: 'Project 2 description',
      es: 'Descripción del proyecto 2',
      pt: 'Descrição do projeto 2'
    },
    tags,
    repo: 'project-2',
    repositoryUrl: 'https://github.com/johndoe/project-2',
    cover: '/assets/projects/project-2.png',
    coverBlurDataURL: '',
    forks: 5,
    stars: 10
  },
  {
    title: 'Project 3',
    description: {
      en: 'Project 3 description',
      es: 'Descripción del proyecto 3',
      pt: 'Descrição do projeto 3'
    },
    tags,
    repo: 'project-3',
    repositoryUrl: 'https://github.com/johndoe/project-3',
    cover: '/assets/projects/project-3.png',
    coverBlurDataURL: '',
    forks: 5,
    stars: 10
  }
]
