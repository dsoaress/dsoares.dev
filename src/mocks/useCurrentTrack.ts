export default jest.mock('@/hooks/useCurrentTrack', () => ({
  useCurrentTrack: () => ({
    title: 'title',
    artist: 'artist',
    album: 'album',
    duration: '05:00'
  })
}))
