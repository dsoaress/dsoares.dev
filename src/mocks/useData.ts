import { config } from '@/data/config'
import type { I18n } from '@/types'

import tailwindConfig from '../../tailwind.config'

type Data = {
  c: typeof config
  t: I18n
}

export const data: Data = {
  c: {
    analytics: {
      domain: 'example.com',
      id: '1111111',
      src: 'https://example.com/analytics.js'
    },
    colors: tailwindConfig.theme.extend.colors,
    locales: {
      defaultLocale: 'en',
      locales: ['en']
    },
    profile: {
      avatar: '/assets/avatar.png',
      favicon: {
        src: '/assets/favicon.png',
        sizes: [16, 32, 48, 64, 96, 128, 256, 512]
      },
      githubUsername: 'johndoe',
      shortTitle: 'JDoe',
      title: 'John Doe'
    },
    skills: [
      'Expo',
      'GraphQL',
      'JavaScript',
      'Jest',
      'Kafka',
      'Nest.js',
      'Next.js',
      'Node',
      'Prisma',
      'React',
      'Redis',
      'Storybook',
      'TypeScript'
    ],
    showResume: true,
    social: [
      { label: 'Dev', url: 'https://dev.to/johndoe' },
      { label: 'Email', url: 'mailto:jdoe@mail.com' },
      { label: 'GitHub', url: 'https://github.com/johndoe' },
      { label: 'Instagram', url: 'https://instagram.com/johndoe' },
      { label: 'LinkedIn', url: 'https://linkedin.com/in/johndoe' },
      { label: 'WhatsApp', url: 'https://watsapp.com/johndoe' }
    ]
  },
  t: {
    description: 'John Doe is a software engineer and a full-stack developer.',
    errors: {
      buttonLabel: 'Go back',
      internalError: 'Internal error',
      notFound: 'Page not found'
    },
    footer: 'Â© 2020 John Doe',
    listeningNow: 'Listening now:',
    projects: {
      description: 'Projects description',
      title: 'Projects'
    },
    resume: {
      file: '/assets/resume.pdf',
      title: 'Resume'
    }
  }
}

export default jest.mock('@/hooks/useData', () => ({
  useData: () => ({ ...data })
}))
