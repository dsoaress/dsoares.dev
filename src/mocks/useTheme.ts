export default jest.mock('next-themes', () => ({
  useTheme: () => ({
    resolvedTheme: 'light',
    setTheme: jest.fn()
  })
}))
