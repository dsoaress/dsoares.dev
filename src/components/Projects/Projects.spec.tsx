import { render } from '@testing-library/react'

import { projects } from '@/mocks'

import { Projects } from '.'

describe('Projects', () => {
  it('should render correctly', () => {
    const { container } = render(<Projects projects={projects} />)
    expect(container).toMatchSnapshot()
  })
})
