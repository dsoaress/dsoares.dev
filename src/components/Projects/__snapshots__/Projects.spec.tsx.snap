// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Projects should render correctly 1`] = `
<div>
  <section
    class="my-20"
    id="projects"
  >
    <h2
      class="m-0 font-bold text-neutral-900 transition-colors duration-300 dark:text-neutral-50 mb-6 text-3xl leading-10"
    >
      Projects
    </h2>
    <p
      class="font-normal text-neutral-900 transition-colors duration-300 dark:text-neutral-50 text-neutral-600 dark:text-neutral-400 text-xl leading-8"
    >
      Projects description
    </p>
    <div
      class="mt-8 grid gap-8 md:grid-cols-2 lg:grid-cols-3"
    >
      <div
        class="relative rounded-md border border-solid border-neutral-200 transition-all duration-300 hover:bg-neutral-200 hover:shadow-md dark:border-neutral-800 dark:hover:bg-neutral-800"
      >
        <a
          aria-label="Project 1"
          class="absolute inset-0 umami--click--project-Project 1"
          href="https://github.com/johndoe/project-1"
          rel="noopener noreferrer"
          target="_blank"
        />
        <div
          class="relative h-64 overflow-hidden rounded-tl-md rounded-tr-md"
        >
          <img
            alt="Project 1"
            src="/assets/projects/project-1.png"
          />
        </div>
        <div
          class="grid gap-4 p-6"
        >
          <h3
            class="m-0 font-bold text-neutral-900 transition-colors duration-300 dark:text-neutral-50 text-lg leading-7"
          >
            Project 1
          </h3>
          <p
            class="font-normal text-neutral-900 transition-colors duration-300 dark:text-neutral-50 text-base leading-6"
          >
            Project 1 description
          </p>
          <footer
            class="flex items-end justify-between"
          >
            <p
              class="font-normal text-neutral-900 transition-colors duration-300 dark:text-neutral-50 text-neutral-600 dark:text-neutral-400 text-sm leading-5"
            >
              tag a, tag b, tag c
            </p>
            <div
              class="ml-3 flex gap-3"
            />
          </footer>
        </div>
      </div>
      <div
        class="relative rounded-md border border-solid border-neutral-200 transition-all duration-300 hover:bg-neutral-200 hover:shadow-md dark:border-neutral-800 dark:hover:bg-neutral-800"
      >
        <a
          aria-label="Project 2"
          class="absolute inset-0 umami--click--project-Project 2"
          href="https://github.com/johndoe/project-2"
          rel="noopener noreferrer"
          target="_blank"
        />
        <div
          class="relative h-64 overflow-hidden rounded-tl-md rounded-tr-md"
        >
          <img
            alt="Project 2"
            src="/assets/projects/project-2.png"
          />
        </div>
        <div
          class="grid gap-4 p-6"
        >
          <h3
            class="m-0 font-bold text-neutral-900 transition-colors duration-300 dark:text-neutral-50 text-lg leading-7"
          >
            Project 2
          </h3>
          <p
            class="font-normal text-neutral-900 transition-colors duration-300 dark:text-neutral-50 text-base leading-6"
          >
            Project 2 description
          </p>
          <footer
            class="flex items-end justify-between"
          >
            <p
              class="font-normal text-neutral-900 transition-colors duration-300 dark:text-neutral-50 text-neutral-600 dark:text-neutral-400 text-sm leading-5"
            >
              tag a, tag b, tag c
            </p>
            <div
              class="ml-3 flex gap-3"
            >
              <span>
                <p
                  class="font-normal text-neutral-900 transition-colors duration-300 dark:text-neutral-50 flex items-center text-neutral-600 dark:text-neutral-400 text-sm leading-5"
                >
                  <svg
                    class="mr-0.5"
                    fill="currentColor"
                    height="1em"
                    stroke="currentColor"
                    stroke-width="0"
                    viewBox="0 0 1024 1024"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M752 100c-61.8 0-112 50.2-112 112 0 47.7 29.9 88.5 72 104.6v27.6L512 601.4 312 344.2v-27.6c42.1-16.1 72-56.9 72-104.6 0-61.8-50.2-112-112-112s-112 50.2-112 112c0 50.6 33.8 93.5 80 107.3v34.4c0 9.7 3.3 19.3 9.3 27L476 672.3v33.6c-44.2 15-76 56.9-76 106.1 0 61.8 50.2 112 112 112s112-50.2 112-112c0-49.2-31.8-91-76-106.1v-33.6l226.7-291.6c6-7.7 9.3-17.3 9.3-27v-34.4c46.2-13.8 80-56.7 80-107.3 0-61.8-50.2-112-112-112zM224 212a48.01 48.01 0 0 1 96 0 48.01 48.01 0 0 1-96 0zm336 600a48.01 48.01 0 0 1-96 0 48.01 48.01 0 0 1 96 0zm192-552a48.01 48.01 0 0 1 0-96 48.01 48.01 0 0 1 0 96z"
                    />
                  </svg>
                  5
                </p>
              </span>
              <span>
                <p
                  class="font-normal text-neutral-900 transition-colors duration-300 dark:text-neutral-50 flex items-center text-neutral-600 dark:text-neutral-400 text-sm leading-5"
                >
                  <svg
                    class="mr-0.5"
                    fill="currentColor"
                    height="1em"
                    stroke="currentColor"
                    stroke-width="0"
                    viewBox="0 0 1024 1024"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M908.1 353.1l-253.9-36.9L540.7 86.1c-3.1-6.3-8.2-11.4-14.5-14.5-15.8-7.8-35-1.3-42.9 14.5L369.8 316.2l-253.9 36.9c-7 1-13.4 4.3-18.3 9.3a32.05 32.05 0 0 0 .6 45.3l183.7 179.1-43.4 252.9a31.95 31.95 0 0 0 46.4 33.7L512 754l227.1 119.4c6.2 3.3 13.4 4.4 20.3 3.2 17.4-3 29.1-19.5 26.1-36.9l-43.4-252.9 183.7-179.1c5-4.9 8.3-11.3 9.3-18.3 2.7-17.5-9.5-33.7-27-36.3zM664.8 561.6l36.1 210.3L512 672.7 323.1 772l36.1-210.3-152.8-149L417.6 382 512 190.7 606.4 382l211.2 30.7-152.8 148.9z"
                    />
                  </svg>
                  10
                </p>
              </span>
            </div>
          </footer>
        </div>
      </div>
      <div
        class="relative rounded-md border border-solid border-neutral-200 transition-all duration-300 hover:bg-neutral-200 hover:shadow-md dark:border-neutral-800 dark:hover:bg-neutral-800"
      >
        <a
          aria-label="Project 3"
          class="absolute inset-0 umami--click--project-Project 3"
          href="https://github.com/johndoe/project-3"
          rel="noopener noreferrer"
          target="_blank"
        />
        <div
          class="relative h-64 overflow-hidden rounded-tl-md rounded-tr-md"
        >
          <img
            alt="Project 3"
            src="/assets/projects/project-3.png"
          />
        </div>
        <div
          class="grid gap-4 p-6"
        >
          <h3
            class="m-0 font-bold text-neutral-900 transition-colors duration-300 dark:text-neutral-50 text-lg leading-7"
          >
            Project 3
          </h3>
          <p
            class="font-normal text-neutral-900 transition-colors duration-300 dark:text-neutral-50 text-base leading-6"
          >
            Project 3 description
          </p>
          <footer
            class="flex items-end justify-between"
          >
            <p
              class="font-normal text-neutral-900 transition-colors duration-300 dark:text-neutral-50 text-neutral-600 dark:text-neutral-400 text-sm leading-5"
            >
              tag a, tag b, tag c
            </p>
            <div
              class="ml-3 flex gap-3"
            >
              <span>
                <p
                  class="font-normal text-neutral-900 transition-colors duration-300 dark:text-neutral-50 flex items-center text-neutral-600 dark:text-neutral-400 text-sm leading-5"
                >
                  <svg
                    class="mr-0.5"
                    fill="currentColor"
                    height="1em"
                    stroke="currentColor"
                    stroke-width="0"
                    viewBox="0 0 1024 1024"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M752 100c-61.8 0-112 50.2-112 112 0 47.7 29.9 88.5 72 104.6v27.6L512 601.4 312 344.2v-27.6c42.1-16.1 72-56.9 72-104.6 0-61.8-50.2-112-112-112s-112 50.2-112 112c0 50.6 33.8 93.5 80 107.3v34.4c0 9.7 3.3 19.3 9.3 27L476 672.3v33.6c-44.2 15-76 56.9-76 106.1 0 61.8 50.2 112 112 112s112-50.2 112-112c0-49.2-31.8-91-76-106.1v-33.6l226.7-291.6c6-7.7 9.3-17.3 9.3-27v-34.4c46.2-13.8 80-56.7 80-107.3 0-61.8-50.2-112-112-112zM224 212a48.01 48.01 0 0 1 96 0 48.01 48.01 0 0 1-96 0zm336 600a48.01 48.01 0 0 1-96 0 48.01 48.01 0 0 1 96 0zm192-552a48.01 48.01 0 0 1 0-96 48.01 48.01 0 0 1 0 96z"
                    />
                  </svg>
                  5
                </p>
              </span>
              <span>
                <p
                  class="font-normal text-neutral-900 transition-colors duration-300 dark:text-neutral-50 flex items-center text-neutral-600 dark:text-neutral-400 text-sm leading-5"
                >
                  <svg
                    class="mr-0.5"
                    fill="currentColor"
                    height="1em"
                    stroke="currentColor"
                    stroke-width="0"
                    viewBox="0 0 1024 1024"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M908.1 353.1l-253.9-36.9L540.7 86.1c-3.1-6.3-8.2-11.4-14.5-14.5-15.8-7.8-35-1.3-42.9 14.5L369.8 316.2l-253.9 36.9c-7 1-13.4 4.3-18.3 9.3a32.05 32.05 0 0 0 .6 45.3l183.7 179.1-43.4 252.9a31.95 31.95 0 0 0 46.4 33.7L512 754l227.1 119.4c6.2 3.3 13.4 4.4 20.3 3.2 17.4-3 29.1-19.5 26.1-36.9l-43.4-252.9 183.7-179.1c5-4.9 8.3-11.3 9.3-18.3 2.7-17.5-9.5-33.7-27-36.3zM664.8 561.6l36.1 210.3L512 672.7 323.1 772l36.1-210.3-152.8-149L417.6 382 512 190.7 606.4 382l211.2 30.7-152.8 148.9z"
                    />
                  </svg>
                  10
                </p>
              </span>
            </div>
          </footer>
        </div>
      </div>
    </div>
  </section>
</div>
`;
